src/datasources/db/v2/postgres-database.service.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  LoggingService,
+  type ILoggingService,
+} from '@/logging/logging.interface';
+import { InjectDataSource } from '@nestjs/typeorm';
+import { Inject, Injectable } from '@nestjs/common';
+import { DataSource, type ObjectLiteral, type Repository } from 'typeorm';
+
+@Injectable()
+export class PostgresDatabaseService {
+  public constructor(
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    @InjectDataSource() private readonly dataSource: DataSource,
+  ) {}
+
+  /**
+   * Fetches the database connection. If the connection is not initialized, it initializes the connection.
+   *
+   * @returns {Promise<DataSource>} The database connection.
+   */
+  public async fetchDatabaseConnection(): Promise<DataSource> {
+    if (!this.dataSource.isInitialized) {
+      this.loggingService.info('PostgresDatabaseService initialized...');
+      await this.dataSource.initialize();
```

Comment:
Do we need to "de-initialize" this `onModuleDestroy`?

Reply:
`NestJS` automatically terminates idle connections on shutdown. I suggest we avoid adding onModuleDestroy and stick with the default behavior.

Reply:
Unrelated to this PR but a question in general: should we then aim to remove other instances of `onModuleDestroy` that fall under this criteria? We should file an issue if so.

Reply:
Could you provide docs regarding that? I'm searching and I cannot find how exactly NestJS/TypeOrm handle the shutdown. We also need to take Kubernetes into consideration, where a kill/reboot of a pod would happen and only a SIGTERM can be received. `OnModuleDestroy` would guarantee proper connection closing in that case. 

Anyway, we can just monitor the opened connections to get confidence around this.

Reply:
@iamacook Well, If they are automatically closed on shut down, we can add an issue to remove the redundant call, As in some cases like here calling the shutdown function manually may cause an error if the connection is already terminated, you need to check whether the connection is active or not before calling the shut down.

Reply:
@hectorgomezv Sure, https://docs.nestjs.com/techniques/database
there is an option that is already deprecated called, `keepConnectionAlive:
`If true, connection will not be closed on the application shutdown (default: false)`

I have also tested the functionality and can confirm that the connection is properly terminated during a graceful shutdown.

We can add `OnModuleDestroy` if you think it's necessary for extra verification, but the connection is already terminated before this hook is triggered.

Reply:
Great, no need then to handle that in the `onModuleDestroy` hook, thanks for clarifying!